### DISCUSSION TITLE
WOW! 🎉😍

### DISCUSSION THREAD

**Discussion by bverkron** [Zavy86/WikiDocs#87] [Show and tell]
Just wanted to drop note to say WOW! I'm so thoroughly impressed by WikiDocs that I almost can't believe it actually exists. I have been hunting for a simple yet feature rich and well designed markdown based wiki platform for weeks and weeks and had been sort of stuck between OtterWiki and flatnotes both of which are great but each lacks some key features the other has and vice versa. Neither 100% satisfied my criteria. WikiDocs has everything I want (basically) with little to no extra "fluff" and a bunch of bonus QoL features that were on my wishlist like mermaid support, image / attachment management, etc. On top of all that it only uses around 12 MB or RAM according to Portainer when it boots up. That's ~10x less than flatnotes and otterwiki!

Kudos on developing a great project. To help share it with more people (and hopefully get more contributors) I [submitted it](https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/456) to the awesome self-hosted GitHub project. Looks like some minor issues with the PR but hopefully those will be resolved soon.

🎉🎉🎉🎉🎉🎉🎉

**Comment by bverkron** [comment]
🟢 = Pro or feature the alternatives do not have
🟠 = Minor issue / annoyance (not deal breaker)
🔴 = Potential deal breaker

#### ⭐️ WikiDocs
* 🟢 Acceptable user management (view code for read access or none for open read access, edit code for editing)
* 🟢 Hierarchy for pages!
* 🟢 Image / attachment management in UI including deleting images from disk! But no rename.
* 🟢 Automatic draft saving every few seconds
* 🟢 Version management and rollback in UI
* 🟢 Active development
* 🟢 Flexible editor views. Preview button, Full screen editor, split editor
* 🟢 Clean print feature built-in
* 🟢 Easy to view and restore previous versions
* 🟢 [Diagram support](https://github.com/Zavy86/WikiDocs/pull/52)
* 🟢 Low RAM usage. Starts at 12MB and only got as high as 40 MB after a ton of editing, mermaid chart renders, etc. flatnotes was 90 MB at startup and otterwiki was over 100 MB
* 🟢 No ToC by default but can add [toc] manually and automatic [might been the way](https://github.com/Zavy86/WikiDocs/issues/80)
* 🟢 Adequate user management which allows for read-only (with / without password) and editing
* ⚪️ Unlimited version history which is good and bad since you cannot limit how much history is kept and have to manually purge older history?
* 🟠 Search is basic. No regex, patterns, etc. Always opens in new tab? Only searches contents of files, not filenames?
* 🟠 No table editing features in WYSIWYG editor like adding rows with buttons or with tab. All manual text manipulation.

#### 🟠 Otterwiki
So close but a few deal breakers or annoyances like risk of losing data, page title case issues, etc.
* 🟢 User management is good (flatnotes has basically 100% admin access or only read access, nothing in between)
* 🟢 Search is robust (supports regex)
* 🟢 Attachment management fairly robust with show, delete, rename, etc
* 🟢 Index page can show page sections which is very handy for easily jumping to info within a page from the index
* 🟢 ToC automatically generated on the right
* 🟢 [Extended syntax](https://otterwiki.com/Examples/All%20Syntax%20Features#special-blocks) for blocks, hidden info, etc.
* 🟢 Version management and rollback in UI?
* 🟠 Requires a commit message every time you save which gets tedious and is not very practical / useful most of the time
* 🟠 Not very active development
* 🟠 Forces Camel Case on page titles. Cannot have titles like "NAS Setup", forces to "Nas Setup"
* 🟠 No hierarchy support (by design?)
* 🟠 No custom sort order in index / menu
* 🟠 No automatic draft saving
* 🔴 Editor can cause data loss ❗ if not careful. If this [#1 issue](https://github.com/redimp/otterwiki/discussions/60) were address then Otter would be good enough to start using.

#### 🟠 flatnotes
* 🟢 Good editor. Preview is a nicer experience than Otter. (Otter can easily cause data loss ❗)
* 🟢 Can sort index by title, last modified, search score
* 🟢 Active development
* 🟠 No ToC
* 🟠 No hierarchy support (by design)
* 🟠 No custom sort order in index / menu

**Comment by bverkron** [comment]
> I [submitted it](https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/456) to the awesome self-hosted GitHub project.

Merged and officially listed @Zavy86 !

https://awesome-selfhosted.net/tags/wikis.html

**Reply by Zavy86** [reply]
thank you so much!
